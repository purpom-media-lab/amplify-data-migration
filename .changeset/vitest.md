---
"@purpom-media-lab/amplify-data-migration": patch
---

Bump vitest from 2.1.8 to 3.0.4
