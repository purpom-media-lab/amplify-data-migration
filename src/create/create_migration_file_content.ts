export const createMigrationClassContent = (
  migrationName: string,
  timestamp: number
) => {
  const className = toPascalCase(migrationName);
  return `import {
  DynamoDBExportKey,
  ExportContext,
  Migration,
  MigrationContext,
  ModelTransformer,
} from "@purpom-media-lab/ampify-data-migration";

export default class ${className}_${timestamp} implements Migration {
  readonly name = "${migrationName}";
  readonly timestamp = ${timestamp};

  // You should implement export method if you want to export data from the table.
  // async export(
  //   context: ExportContext
  // ): Promise<Record<string, DynamoDBExportKey>> {
  //   // Export table to S3 bucket with Point-in-Time Recovery
  //   const key = await context.modelClient.exportModel("<Your Model name>");
  //   return { <Your Model name>: key };
  // }

  async run(context: MigrationContext) {
    // type OldModel = {};
    // type NewModel = {};
    // const transformer: ModelTransformer<OldModel, NewModel> = async (
    //   oldModel
    // ) => {
    //   return oldModel;
    // };
    // If you run import exported data to new a table, you can use this code.
    // await context.modelClient.runImport("<Your Model name>", transformer);
    // If you run update table, you can use this code.
    // await context.modelClient.updateTable("<Your Model name>", transformer);
  }
}
`;
};

const toPascalCase = (str: string) =>
  str
    .match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g)
    ?.map((x: string) => x.slice(0, 1).toUpperCase() + x.slice(1).toLowerCase())
    ?.join("");
