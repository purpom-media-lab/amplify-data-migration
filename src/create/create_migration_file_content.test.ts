import { createMigrationClassContent } from "./create_migration_file_content";
import { describe, test, expect } from "vitest";

describe("createMigrationFileContent", () => {
  test("should return Migration file content", () => {
    const name = "test_migration";
    const timestamp = 1726384783922;
    const result = createMigrationClassContent(name, timestamp);
    expect(result).toMatchSnapshot();
  });
});
