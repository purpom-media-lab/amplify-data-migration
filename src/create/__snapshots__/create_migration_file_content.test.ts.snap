// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createMigrationFileContent > should return Migration file content 1`] = `
"import {
  DynamoDBExportKey,
  ExportContext,
  Migration,
  MigrationContext,
  ModelTransformer,
} from "@purpom-media-lab/amplify-data-migration";

export default class TestMigration_1726384783922 implements Migration {
  readonly name = "test_migration";
  readonly timestamp = 1726384783922;

  // You should implement export method if you want to export data from the table.
  // async export(
  //   context: ExportContext
  // ): Promise<Record<string, DynamoDBExportKey>> {
  //   // Export table to S3 bucket with Point-in-Time Recovery
  //   const key = await context.modelClient.exportModel("<Your Model name>");
  //   return { <Your Model name>: key };
  // }

  async run(context: MigrationContext) {
    // type OldModel = {};
    // type NewModel = {};
    // const transformer: ModelTransformer<OldModel, NewModel> = async (
    //   oldModel
    // ) => {
    //   return oldModel;
    // };
    // If you run import exported data to new a table, you can use this code.
    // await context.modelClient.runImport("<Your Model name>", transformer);
    // If you run update table, you can use this code.
    // await context.modelClient.updateTable("<Your Model name>", transformer);
  }
}
"
`;
